### PR TITLE
[#195] Add jwt-cpp

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Addresses #195

### CODE REVIEW

**Comment by SwooshyCueb** [comment]
I've changed the version to a sha to get rid of a VLA.
Additionally, the cmake config files end up in the right place now.

Builds fine, but otherwise currently untested. I will probably end up having to write a find module for it like for nlohmann-json. Since jwt-cpp isn't likely to be used outside of plugins, this find module may end up *in the externals package itself*.

**Comment by SwooshyCueb** [comment]
I think this is ready now

**Review by alanking — APPROVED** [review]
Looks good to me. Maybe one more approval before #

File: versions.json [hunk 1]
```diff
@@ -167,6 +167,24 @@
         ],
         "fpm_directories": ["include","lib"]
     },
+    "jwt-cpp": {
+        "git_repository": "https://github.com/Thalhammer/jwt-cpp",
```
Comment (trel) [inline versions.json:171]: we'd rather build from our own fork.

Comment (trel) [reply versions.json:171]: now at https://github.com/irods/jwt-cpp

Comment (SwooshyCueb) [reply versions.json]: Didn't know we had one yet. Fixed

File: versions.json [hunk 2]
```diff
@@ -167,6 +167,23 @@
         ],
         "fpm_directories": ["include","lib"]
     },
+    "jwt-cpp": {
+        "enable_sha": true,
```
Comment (trel) [inline versions.json:171]: so i think this is not necessary since we now have a fork in the irods namespace?    please try without it.

Comment (SwooshyCueb) [reply versions.json:171]: This is needed because we're not using a tag or branch, but a commit sha to identify what to build.

Comment (trel) [reply versions.json:171]: oh, i see.   it's not a 'real' tag by them.   it's pre-0.7.0.

**Review by korydraughn — APPROVED** [review]
Looks good. Add pound when ready.

**Comment by SwooshyCueb** [comment]
\#'d

**Pull request merged**